app.py

```diff
@@ -40,9 +35,20 @@
 
 # Mastodon
 try:
+    client_file = param['mastodon']['client_id_file']
+    if not os.path.exists(client_file):
+        raise Exception("File not found: " + client_file)
+    access_token_file = param['mastodon']['access_token_file']
+    if not os.path.exists(access_token_file):
+        raise Exception("File not found: " + client_file)
+
+    mastodon_url = param['mastodon'].get('url', 'https://mastodon.social')
+    print("Using Mastodon instance at:", mastodon_url)
+
     mastodon = Mastodon(
-        client_id=param['mastodon']['client_id_file'],
-        access_token=param['mastodon']['access_token_file']
+        client_id=client_file,
+        access_token=access_token_file,
+        api_base_url=mastodon_url
```

Comment:
Added this after discovering it refused to work for anything but the flagship instance.